src/Infrastructure.EntityFramework/AdminConsole/Repositories/OrganizationRepository.cs

```diff
@@ -403,4 +403,41 @@ public async Task<OrganizationSeatCounts> GetOccupiedSeatCountByOrganizationIdAs
             };
         }
     }
+
+    public async Task<IEnumerable<Core.AdminConsole.Entities.Organization>> GetOrganizationsForSubscriptionSyncAsync()
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        await using var dbContext = GetDatabaseContext(scope);
+
+        var organizations = await dbContext.Organizations
+            .Where(o => o.SyncSeats == true && o.Seats != null)
+            .ToArrayAsync();
+
+        return organizations;
+    }
+
+    public async Task UpdateSuccessfulOrganizationSyncStatusAsync(IEnumerable<Guid> successfulOrganizations, DateTime syncDate)
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        await using var dbContext = GetDatabaseContext(scope);
+
+        await dbContext.Organizations
+            .Where(o => successfulOrganizations.Contains(o.Id))
+            .ExecuteUpdateAsync(o => o
+                .SetProperty(x => x.SyncSeats, false)
+                .SetProperty(x => x.RevisionDate, syncDate.Date));
+    }
+
+    public async Task IncrementSeatCountAsync(Guid organizationId, int increaseAmount, DateTime requestDate)
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        await using var dbContext = GetDatabaseContext(scope);
+
+        await dbContext.Organizations
+            .Where(o => o.Id == organizationId)
+            .ExecuteUpdateAsync(s => s
+                .SetProperty(o => o.Seats, o => o.Seats + increaseAmount)
+                .SetProperty(o => o.SyncSeats, true)
```

Comment:
here we're incrementing and setting sync seats to true.

Reply:
Gotcha, got confused in the diff view.